### ISSUE TITLE
BUG: Cluster: rebooted device loses some settings

### ISSUE THREAD

**Issue by HablasNapa** [rameplayerorg/rameplayer-defects#32]
Scenario: Cluster is playing the assigned playlist and one of the rameplayers is rebooted. Rest of the cluster finishes playing the assigned file.

Command of **play** is given from cluster GUI.

Result: Other rameplayers move on to the next file in their playlist except for the rebooted one which starts to play the first file in directory.

It appears that this result comes when a device rejoins the cluster after reboot and receives the **play** command but does not retain the original playlist.

**Comment by viliman** [comment]
Yes this is know problem and missing feature from backend it doesn't store the playlist at the moment

**Comment by jlauha** [comment]
Mid-development testing report, using current development versions of backend & webui & utils:

If cluster player is rebooted (backend restart), controlling player's webui will correctly notice the player missing (red toaster), and will also notice when it comes back online (red toaster goes away). 

After this the Synchronized playlist view looks like "everything is normal": cluster player playlist visible just as one would expect to be (now that the cluster drop/reconnect is detected, and also playlists are now saved).

However a few issues remain:
- cursor of the cluster player is not re-initialized automatically to what's the correct position in synced playlist (it will only point automatically to the 1st media item; so if one actually plays file, the cluster player will erroneously just always play that 1st item)
- any change in the controlling player's (master playlist) cursor won't affect the cluster player cursor at all

Note: If controlling player's webui is reloaded/refreshed at some point after the cluster player was restarted, then one can see that connection to the synchronized playlist has actually been lost.

---

So just for additional info -- below is the current steps which allow properly re-attaching a cluster player to the cluster:
- after it has been booted, use the X button in controlling player webui to remove the particular cluster player's synced playlist from the cluster
- use Synchronize option again to add the same player unit back to the cluster (with the same playlist selection)
- reorder any extra "empty slots" in the cluster player's playlist again to the correct positions, if any.
- then, to match current playlist cursor position in cluster player: click cursor to some other media item in the list in controlling player's playlist, and then click again to the position you were in first. (two clicks since one can't click the same item again to re-set cursor to that item)

**Comment by viliman** [comment]
Great report @jlauha

**Comment by jlauha** [comment]
Also tried this:
- save working cluster configuration to a local file (json)
- verified importing it works in another browser
- rebooted cluster player
- imported the cluster configuration json for use
  Result: synced playlist of the rebooted player shows up as empty playlist in the cluster.

**Comment by iqqmuT** [comment]
Good thorough work @jlauha!
Backend is not persisting playlist item ids yet and that might be the root problem. Let's wait for backend changes and try again.
